features/323-login-id/api.md

```diff
@@ -0,0 +1,170 @@
+    },
+    "access_token": "..."
+}
+```
+
+## Use Case Examples
+
+Unless specified, following examples assume using default configuration.
+
+### Signup with single login ID
+```typescript
+const user1 = await signupWithEmail("test@example.com", "12345678");
+// user1.metadata is { "email": "test@example.com" }
+const user2 = await signup({ phone: "+85299999999" }, "12345678");
+// user2.metadata is { "phone": "+85299999999" }
+
+// following should fail:
+await signup({ fingerprint: "ZmluZ2VycHJpbnQ=" }, "12345678"); // not white-listed
+```
+
+### Signup with realms
+Assume `teacher` and `student` are whitelisted realms.
+```typescript
+await signupWithEmail("test@example.com", "12345678", undefined, "teacher");
+await createLoginID("email", "test@example.com", "student");
+// then:
+
+// - following should succeed:
+await loginWithEmail("test@example.com", "12345678", "teacher");
+await loginWithEmail("test@example.com", "12345678", "student");
+
+// - following should fail:
+await loginWithEmail("test@example.com", "12345678"); // realm not matched
+await signupWithEmail("test@example.com", "12345678", undefined, "admin"); // cannot create login ID (use createLoginID instead)
+await signupWithEmail("test@example.com", "12345678"); // cannot reuse login ID for new user
+```
+
+### Signup with multiple login ID
+Assume `email` and `username` are white-listed login ID keys,
+with exactly one username allowed.
+
+```typescript
+const user = await signup([
+    {"email": "test@oursky.com"},
+    {"email": "test@example.com"},
+    {"username": "test"}
+], "12345678");
+// user.metadata is { "email": "test@oursky.com", "username": "test" }
+// then:
+
+// - following should succeed:
+await loginWithEmail("test@oursky.com", "12345678");
+await login("test@example.com", "12345678");
+await loginWithUsername("test", "12345678");
+
+// - following should fail:
+await login({
+    "email": "test@oursky.com",
+    "username": "test"
+}, "12345678"); // cannot login with more than 2 login ID
+await signupWithEmail("test@example.com", "12345678"); // cannot reuse login ID
+await signup([
+    {"email": "test1@example.com"},
+    {"email": "test2@example.com"}
+], "12345678"); // username is not present
+```
+
+### User verification
+Assume `email` and `username` are white-listed login ID keys,
+with maximum 2 emails.
+Assume verification criteria is set as `all`.
+
+```typescript
+let user = await signup([
+    {"email": "test@oursky.com"},
+    {"email": "test@example.com"},
+    {"username": "test"}
+]);
+// user.verified is false
+// user.verify_info is {}
+
+// following should fail:
+await requestVerification("username", "test"); // unsupported login ID key
+await requestVerification("email", "test1@oursky.com"); // login ID does not belong to current user
+
+// following should succeed:
+await requestVerification("email", "test@oursky.com");
+
+// after verification:
+user = await whoami();
+// user.verified is false
```

Comment:
I think it is more common use case for at least ONE loginID is verified, the account is considered verified? (We can provide a configuration, but I think the default should be ONE or more login ID verified => verified is true.

Reply:
There is a configuration for 'any'/'all' emails. However, there is no default value. I'll set it to 'any' if you prefer.

Reply:
I think it needs a default value? If it is not any / all not sure what the verified meant :)

yea I prefer any, but that's because it seems more common? (Most services I used, if you have ONE email address verified, you can act as a verified user)